.github/workflows/jacktrip.yml

```diff
@@ -597,13 +580,105 @@ jobs:
           path: ${{ env.BUILD_PATH }}/${{ matrix.bundle-path }}
       - name: upload installer
         uses: actions/upload-artifact@v2
-        if: matrix.installer-path && (!startsWith(github.ref, 'refs/tags/') || (startsWith(github.ref, 'refs/tags/') && matrix.release-name)) && steps.check-secrets.outputs.available != 'true'
+        if: matrix.installer-path && (!startsWith(github.ref, 'refs/tags/') || (startsWith(github.ref, 'refs/tags/') && matrix.release-name)) && (steps.check-secrets.outputs.available != 'true' || runner.os == 'Windows')
+        with:
+          name: JackTrip-${{ steps.set-version.outputs.version }}-${{ steps.set-version.outputs.name }}-installer
+          path: ${{ env.BUILD_PATH }}/${{ matrix.installer-path }}
+  check-secrets:
+    name: Check if secrets exist to determine if build & sign run
+    runs-on: ubuntu-latest
+    outputs:
+      should_run: ${{ steps.setvar.outputs.should_run }}
+    steps:
+      - id: setvar
+        run: |
+          if [[ "${{ secrets.APP_SIGNING_CERT_BASE64 }}" != "" && \
+                "${{ secrets.INSTALLER_SIGNING_CERT_BASE64 }}" != "" && \
+                "${{ secrets.CERT_PASSPHRASE }}" != "" && \
+                "${{ secrets.APP_CERT_NAME }}" != "" && \
+                "${{ secrets.INSTALLER_CERT_NAME }}" != "" && \
+                "${{ secrets.APPLE_APP_SPECIFIC_PWD }}" != "" && \
+                "${{ secrets.TOTP_SECRET }}" != "" && \
+                "${{ secrets.ESIGNER_CREDENTIAL_ID }}" != "" && \
+                "${{ secrets.SSL_COM_PWD }}" != "" && \
+                "${{ secrets.KEYCHAIN_PWD }}" != "" ]]; \
+          then
+            echo "Secrets to sign artifacts were configured in the repo"
+            echo "should_run=true" >> $GITHUB_OUTPUT
+          else
+            echo "Secrets to sign artifacts were not configured in the repo"
+            echo "should_run=false" >> $GITHUB_OUTPUT
+          fi
+  sign:
+    needs: [build, check-secrets]
+    if: needs.check-secrets.outputs.should_run == 'true'
+    runs-on: ${{ matrix.runs-on }}
+    name: ${{ matrix.name }}
+    strategy:
+      fail-fast: false # don't abort if one of the build false
+      matrix:
+        include:
+          - name: Sign Windows artifacts
+            release-name: Windows-x64
+            build-job-name: Windows-x64-qmake-gcc-static-bundled_rtaudio
+            runs-on: ubuntu-latest
+            binary-path: binary
+            installer-path: installer
```

Comment:
technically we don't have to use a matrix here, since this will only be done for Windows; but it can stay

Reply:
Yeah i was pulling it out of other code